components/bl-cascadeSelect-component/src/index.js

```diff
@@ -0,0 +1,68 @@
+import { useState, useEffect, useCallback, useRef } from 'react';
+import { CollapseButtonIcon, Cascade } from './subcomponent';
+import { openCascade, validate } from './helpers';
+
+const { cn } = BackendlessUI.CSSUtils;
+
+export default function CascadeSelect({ component, eventHandlers }) {
+  const { display, classList, style, cascade, placeholder } = component;
+  const { onClickItem } = eventHandlers;
+
+  const [itemsCascade, setItemsCascade] = useState();
```

Comment:
what's the difference between them?
- itemsCascade
- parentItems
- items

Reply:
- `itemsCascade` is a validated object which comes from property `cascade`
- `parentItems` is the array of arrays of parent items(items which have children) separated by `levelOfNesting`
- `items` is the array of items which we can select(which haven't children)

Reply:
I still do not understand why we need 3 lists instead of a single one `itemsCascade`
we can validate/transform the source list into a list that suits all needs, the target list can have any structure we need and items can have all necessary information about `clickable/nestingLevel/etc`

Reply:
because it's easy to find items in an array, instead of an array of objects that have in turn, have their own array of objects and so on to infinity

Reply:
parent reference should solve that problem